### PR TITLE
Fix Issue 70 add file mode parameters

### PR DESCRIPTION
**Pull request by TJM**
#### Pull Request (PR) description
Add parameters for file modes instead of hardcoding them. Default to "Red Hat" RPM permissions (for now).

#### This Pull Request (PR) fixes the following issues
Fixes Issue #70

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+  Integer[0,59] $cron_hourly_minute      = $logrotate::params::cron_hourly_minute,
+  String $cron_hourly_file               = $logrotate::params::cron_hourly_file,
+  Stdlib::Filemode $cron_file_mode       = $logrotate::params::cron_file_mode,
+  String $configdir                      = $logrotate::params::configdir,
+  String $logrotate_bin                  = $logrotate::params::logrotate_bin,
+  String $logrotate_conf                 = $logrotate::params::logrotate_conf,
+  Stdlib::Filemode $logrotate_conf_mode  = $logrotate::params::logrotate_conf_mode,
+  Boolean $manage_package                = $logrotate::params::manage_package,
+  String $rules_configdir                = $logrotate::params::rules_configdir,
+  Stdlib::Filemode $rules_configdir_mode = $logrotate::params::rules_configdir_mode,
```
Comment (TJM) [inline manifests/init.pp]: This is why I dislike lining up equals (and why I jokingly asked if we could just change to module level hieradata :)

File: metadata.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.13.1 <5.0.0"
+      "version_requirement": ">= 4.22.0 < 5.0.0"
```
Comment (TJM) [inline metadata.json]: This appears to be the version of stdlib where they added Stdlib::Filemode

File: manifests/init.pp [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+  Optional[Hash] $config                 = undef,
+  Integer[0,23] $cron_daily_hour         = $logrotate::params::cron_daily_hour,
+  Integer[0,59] $cron_daily_minute       = $logrotate::params::cron_daily_minute,
+  Integer[0,59] $cron_hourly_minute      = $logrotate::params::cron_hourly_minute,
+  String $cron_hourly_file               = $logrotate::params::cron_hourly_file,
+  Stdlib::Filemode $cron_file_mode       = $logrotate::params::cron_file_mode,
+  String $configdir                      = $logrotate::params::configdir,
+  String $logrotate_bin                  = $logrotate::params::logrotate_bin,
+  String $logrotate_conf                 = $logrotate::params::logrotate_conf,
+  Stdlib::Filemode $logrotate_conf_mode  = $logrotate::params::logrotate_conf_mode,
```
Comment (juniorsysadmin) [inline manifests/init.pp]: You probably knew this already but be aware that `Stdlib::Filemode` does not support all of the variants of `File mode =>`. For instance, `a+rwx`

Comment (TJM) [reply manifests/init.pp]: Correct, it currently looks like: `Pattern[/^[0124]{1}[0-7]{3}$/]` ... which matches the permissions that are set in params. 


I think the right thing to do would be to improve Stdlib::Filemode, if someone wants to use symbolic file modes. If the folks here feel strongly that we need to support it before that time, we could just make the type `String[3]` or whatever. I don't feel like we should make our own FileMode to support it.

**Pull request merged**